substrate/primitives/metadata-ir/src/types.rs

```diff
@@ -340,19 +343,29 @@ impl IntoPortable for StorageEntryTypeIR {
 pub struct PalletCallMetadataIR<T: Form = MetaForm> {
 	/// The corresponding enum type for the pallet call.
 	pub ty: T::Type,
+	/// Deprecation status of the pallet call itself
+	pub deprecation_info: DeprecationStatus<T>,
+	/// Deprecation status of the call indexes
+	pub deprecated_indexes: BTreeMap<u8, DeprecationStatus<T>>,
```

Comment:
We could use `Compact<u8>` instead of `u8` to be future proof. Same for the other enumeration types.

Reply:
pushed this change to #4994